File: src/test/console/java/src/main/java/web/IndexPage.java
package web;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

public class IndexPage extends Base {
  @FindBy(xpath= "//*[@id=\"content\"]/h1[text() = \"Welcome to the Internet\"]")
  public WebElement header;
  @FindBy(xpath = "//*[@id=\"content\"]/ul/li/a[text()=\"Dynamic Controls\"]")
  public WebElement elementByXpath;
  @FindBy(css = "#content > ul > li:nth-child(11) > a")
  public WebElement elementByCss;
  @FindBy(tagName = "a")
  public WebElement elementByTagName;
  @FindBy(linkText = "Dynamic Controls")
  public WebElement elementByLinkText;
  @FindBy(partialLinkText = "Dynamic Controls")
  public WebElement elementByPartialLinkText;

  public IndexPage(WebDriver driver, long timeoutInSeconds) {
    super(driver, timeoutInSeconds, "http://the-internet.herokuapp.com");
  }

  public void open() throws Exception {
    super.open();
    this.waitForElementVisible(header);
  }

  public Boolean existElementByXpath() {
    return elementByXpath.isDisplayed();
  }

  public Boolean existElementByCss() {
    return elementByCss.isDisplayed();
  }

  public Boolean existElementByTagName() {
    return elementByTagName.isDisplayed();
  }

  public Boolean existElementByLinkText() {
    return elementByLinkText.isDisplayed();
  }

  public Boolean existElementByPartialLinkText() {
    return elementByPartialLinkText.isDisplayed();
  }

}

File: src/test/console/dotnet/Jsonwired/web/IndexPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace web {
	public class IndexPage : Base {
  		// Page elements
		[FindsBy (How = How.XPath, Using = "//*[@id=\"content\"]/h1[text() = \"Welcome to the Internet\"]")]
		public IWebElement header ;
		[FindsBy (How = How.XPath, Using = "//*[@id=\"content\"]/ul/li/a[text()=\"Dynamic Controls\"]")]
		public IWebElement elementByXpath ;
		[FindsBy(How = How.CssSelector, Using = "#content > ul > li:nth-child(11) > a")]
		public IWebElement elementByCss ;
		[FindsBy(How = How.TagName, Using = "a")]
		public IWebElement elementByTagName;
		[FindsBy(How = How.LinkText, Using = "Dynamic Controls")]
		public IWebElement elementByLinkText ;
		[FindsBy (How =How.PartialLinkText, Using = "Dynamic Controls")]
		public IWebElement elementByPartialLinkText;

		public IndexPage(IWebDriver driver, long timeoutInSeconds) :
			base(driver, timeoutInSeconds, "http://the-internet.herokuapp.com") {
			}

		public override void open() {
			base.open();
			this.waitForElementVisible(header);
		}
	}
}
